packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,135 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { getSpacingStyles, Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  MarginProps,
+  PaddingProps,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+export type IconSize = 'large' | 'medium' | 'small' | 'tiny';
+
+export type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+
+const sizes: { [key in IconSize]: { [key in 'height' | 'width']: string } } = {
+  large: {
+    height: '32px',
+    width: '32px',
+  },
+  medium: {
+    height: '24px',
+    width: '24px',
+  },
+  small: {
+    height: '18px',
+    width: '18px',
+  },
+  tiny: {
```

Comment:
@denkristoffer I think as a follow-up, we could re-visit the sizes for Icon component, I think we should consider `xsmall` namespace instead of tiny. WDYT? (not a blocker, just a thought)

Reply:
I think that's a good idea. The old discussion seemed unfinished to me, so I would say we leave it for a follow-up PR.

Reply:
@denkristoffer This is aligned with design now, can we please have these [sizes](https://www.figma.com/file/mw7H4EffGQAR3FXnUu6HQS/Outdated-Campaigns-UI-Kit?node-id=1281%3A711) in place? should be mapped with spacing values as well

Reply:
Should we rename to follow the token names then? But then the smallest icon size would be called "medium" 🤔😕

Reply:
Nevermind the tokens wouldn't work until we introduce one for 20px. I'll just hardcode the sizes for now.

Reply:
@mshaaban0 I'd like to push the size changes to a later PR so that these implementation changes don't get tangled up with changes to tokens and style.